dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/EnrollmentSupplier.java

```diff
@@ -60,11 +59,9 @@ public void preheatAdd(TrackerObjects trackerObjects, TrackerPreheat preheat) {
     if (programsWithoutRegistration.isEmpty()) {
       programsWithoutRegistration = programStore.getByType(ProgramType.WITHOUT_REGISTRATION);
     }
+
     if (!programsWithoutRegistration.isEmpty()) {
-      List<Enrollment> enrollments =
-          DetachUtils.detach(
```

Comment:
Is that not still needed? or are we not getting a proxied enrollment with the `EntityManager`?

Reply:
Agreed. I think we still need to detach the enrollments.

Reply:
yep, nice catch!